linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/http/ResponseFramingFilter.scala

```diff
@@ -0,0 +1,44 @@
+package io.buoyant.linkerd.protocol.http
+
+import com.twitter.finagle._
+import com.twitter.finagle.http.{Request, Response}
+import com.twitter.util.Future
+import io.buoyant.linkerd.protocol.FramingException
+
+/**
+ * A filter that fails badly-framed responses
+ */
+class ResponseFramingFilter extends SimpleFilter[Request, Response] {
+  override def apply(
+    request: Request,
+    service: Service[Request, Response]
+  ): Future[Response] =
+    service(request) flatMap { response =>
+      val numHeaders = response.headerMap.getAll("Content-Length").toSet.size
+      if (numHeaders > 1) {
+        // if the response has multiple Content-Length headers with unique
+        // values, it's invalid
+        Future.exception(FramingException(
```

Comment:
(I'm calling it `ProtocolException` rather than `ProtocolError`, since IIRC the Java/Scala convention is for `Error`s to be fatal errors, while `Exception`s are recoverable)